subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -336,6 +345,18 @@ int zsock_getaddrinfo(const char *host, const char *service,
 		*res = NULL;
 	}
 	return ret;
+#endif
+
+	return DNS_EAI_FAIL;
+}
+
+void zsock_freeaddrinfo(struct zsock_addrinfo *ai)
+{
+#if defined(CONFIG_NET_SOCKETS_OFFLOAD)
+	return socket_offload_freeaddrinfo(ai);
+#endif
+
+	free(ai);
```

Comment:
I wonder if we may get a warning/error from some proganalysis tool with such code ;-).

Reply:
Good catch, very likely that sooner or later it'll show up as one of the "unreachable code" warnings. I'll rework this to `if (IS_ENABLED(CONFIG_NET_SOCKETS_OFFLOAD)) { ... }` construct.

Anyway, I'm still wondering if there's a better way to tackle DNS offloading, current solution is no different from what we had so far. Or perhaps it's good enough for now, reworking it in the future with better approach should have much less impact than the current offloading engine rework.

Reply:
I've applied the change I've mentioned. Unless we misunderstood each other it's here.

Reply:
Ah, ok. I think in terms of #ifdef, so was looking for:
~~~
#ifdef CONFIG_NET_SOCKETS_OFFLOAD
return socket_offload_freeaddrinfo(ai);
#else
free(ai);
#endif
~~~

And then similar if-else translation. I'd say it's more clear this way, but ok, this one would work either indeed.